src/pathfinder/yapf/yapf_common.hpp

```diff
@@ -119,6 +119,32 @@ class CYapfT
 {
 };
 
-
+/**
+ * Calculates the octile distance cost between the current node and a destination tile.
+ * @tparam TNode Type of the starting node.
+ * @param start_node Node used to get the starting tile / direction.
+ * @param destination_tile Destination tile.
+ * @return Octile distance cost between starting node and destination tile.
+ */
+template <typename TNode>
+int OctileDistanceCost(TNode &start_node, TileIndex destination_tile)
+{
+	static constexpr int dg_dir_to_x_offs[] = {-1, 0, 1, 0};
+	static constexpr int dg_dir_to_y_offs[] = {0, 1, 0, -1};
+
+	const TileIndex tile = start_node.GetTile();
+	const DiagDirection exitdir = TrackdirToExitdir(start_node.GetTrackdir());
```

Comment:
For road vehicles and ships, `GetTile()` and `GetTrackdir()` are not used:
```
		TileIndex tile = n.segment_last_tile;
		DiagDirection exitdir = TrackdirToExitdir(n.segment_last_td);
```
So I'm sure if this new code actually equivalent.